docs/limitations.md

```diff
@@ -119,10 +126,56 @@ the impact on simulation results.
 is more accurate, since syscalls on real systems *do* take non-zero time, but it
 makes the time model more complex to understand and reason about.
 
-* It still doesn't account for time spent by the CPU executing code,
-which also means that a busy-loop that makes no syscalls at all can still lead
-to deadlock. Fortunately such busy loops are rare and are generally agreed upon
-to be bugs, since they'd also potentially monopolize a CPU indefinitely when run
-natively.
+### Workaround: have Shadow preempt CPU-only busy-loops{#cpu-busy-loops}
+
+In cases where enabling `--model-unblocked-syscall-latency` doesn't get the
+simulation out of the busy loop, it may be because the busy-loop makes no syscalls or
+time-checks at all, and instead is waiting indefinitely for another thread to
+modify memory (e.g. to flip a "ready" flag). In Shadow's default mode of
+operation, it will never regain control from such loops, and hence can't move
+the simulation forward.
+
+Such loops can be escaped by enabling the experimental option
+`--native-preemption-enabled`. In this mode of operation, Shadow uses a native
+Linux timer to preempt the thread in such situations, moving simulated time
+forward and allowing other threads in the process to run.
+
+Drawbacks:
+
+* Loss of simulation determinism. In different runs of the simulation, the native timer
+  may fire at different times. In cases where it is truly interrupting a
+  CPU-only-busy-loop that would otherwise run indefinitely, this is unlikely to
+  ultimately change the results of the simulation, but *could*, e.g. if the busy
+  loop counts how many times it iterates and that count is used later. Worse,
+  the native timer may fire during long CPU-only operations that *would* finish
+  on their own (e.g. copying, encrypting, or decrypting large chunks of data in
+  memory), firing at different points in some runs of the simulation, or not at
+  all in some runs of the simulation.
+
+* It may cause the simulation to run slower.
+
+  * As with `--model-unblocked-syscall-latency`, when this feature actually does
+    cause a thread to be rescheduled, there is some performance overhead for that
+    rescheduling.
+
+  * Even when the timer never fires, there is some additional overhead for the
+    additional bookkeeping and management of the native timers. This overhead
+    *usually* won't be very noticeable other than in managed code that makes
```

Comment:
Note to self to update this: this is definitely not true; the overhead is substantial.

Reply:
I think when I wrote it I was thinking about the overhead of the feature when not enabled. When it is enabled it introduces about 4 syscalls to every managed-code-syscall (or other transition from managed code to shim code and back), so unsurprisingly has noticeable overhead.

Reply:
Can we add a note about the substantial overhead to the CLI help string (if we didn't already) so that if a user sees the help menu and then enables the switch we'll have tempered their expectations WRT performance?